samples/user-interface/appwidgets/src/main/java/com/example/platform/ui/appwidgets/glance/weather/WeatherGlanceWidget.kt

```diff
@@ -72,23 +75,34 @@ class WeatherGlanceWidget : GlanceAppWidget() {
         private val largeMode = DpSize(260.dp, 280.dp)
     }
 
-    // Override the state definition to use our custom one using Kotlin serialization
-    override val stateDefinition = WeatherInfoStateDefinition
-
     // Define the supported sizes for this widget.
     // The system will decide which one fits better based on the available space
     override val sizeMode: SizeMode = SizeMode.Responsive(
         setOf(thinMode, smallMode, mediumMode, largeMode)
     )
 
+
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    override suspend fun provideGlance(context: Context, id: GlanceId) {
+        provideContent { Content() }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
     @Composable
-    override fun Content() {
+    fun Content() {
         // Get the stored stated based on our custom state definition.
-        val weatherInfo = currentState<WeatherInfo>()
+        val weatherInfo by WeatherRepo.currentWeather.collectAsState()
         // It will be one of the provided ones
         val size = LocalSize.current
+
+        //Create the update weather on click lambda
+        val context = LocalContext.current;
+        val scope = rememberCoroutineScope()
+        val updateWeather : () -> Unit = {scope.launch { refreshWeather( context ) }};
```

Comment:
Nit: format this line using the AS formatter. Also run the spotless to fix any issue
```
./gradlew --init-script gradle/init.gradle.kts spotlessApply
```